textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -256,24 +273,50 @@ private ITuple joinTuples(ITuple outerTuple, ITuple innerTuple, JoinPredicate jo
 			return null;
 		}
 
-		// TODO Discuss and implement as to what Schema has to be.
-		// It shouldn't just be inner tuple or outer tuple schema.
-		Schema schema = innerTuple.getSchema();
-		List<IField> fieldList = innerTuple.getFields();
-		IField[] nextTupleField = new IField[fieldList.size()];
+		List<IField> newFieldList = new ArrayList<>();
+
+		ArrayList<Attribute> innerAttrList = new ArrayList<>();
+		innerAttrList.addAll(innerTuple.getSchema().getAttributes());
+		ArrayList<Attribute> outerAttrList = new ArrayList<>();
+		outerAttrList.addAll(outerTuple.getSchema().getAttributes());
+
+		List<Attribute> newAttrList = new ArrayList<>();
+		Iterator<Attribute> innerAttrListIter = innerAttrList.iterator();
+
+		while (innerAttrListIter.hasNext()) {
+			Attribute nextAttr = innerAttrListIter.next();
+			if (!nextAttr.equals(SchemaConstants.SPAN_LIST_ATTRIBUTE)
+					&& outerAttrList.contains(nextAttr)) {
+				String fieldName = nextAttr.getFieldName();
+				if (outerTuple.getField(fieldName)
+						.equals(innerTuple.getField(fieldName))) {
+					newAttrList.add(nextAttr);
+					IField nextField = outerTuple.getField(fieldName);
+					newFieldList.add(nextField);
+				}
+			}
+		}
 
-		for (int index = 0; index < nextTupleField.length - 1; index++) {
-			nextTupleField[index] = fieldList.get(index);
+		if (newAttrList.isEmpty() || 
+				!newAttrList.contains(joinPredicate.getjoinAttribute())) {
```

Comment:
If the join attribute is in both the inner schema and outer schema, then I think "newAttrList.contains(joinPredicate.getjoinAttribute)" is always true, right?  If so, the second disjunction is not needed.